packages/wdio-testingbot-service/src/launcher.ts

```diff
@@ -19,23 +19,32 @@ export default class TestingBotLauncher {
             return
         }
 
-        const tbTunnelIdentifier = `TB-tunnel-${Math.random().toString().slice(2)}`
+        const tbTunnelIdentifier = this.options.tbTunnelOpts && this.options.tbTunnelOpts.tunnelIdentifier ? this.options.tbTunnelOpts.tunnelIdentifier : `TB-tunnel-${Math.random().toString().slice(2)}`
 
         this.tbTunnelOpts = Object.assign({
             apiKey: config.user,
             apiSecret: config.key,
             'tunnel-identifier': tbTunnelIdentifier,
         }, this.options.tbTunnelOpts)
 
-        let tbOptions = capabilities['tb:options']
-        if (!tbOptions) {
-            tbOptions = {}
-        }
+        if (Array.isArray(capabilities)) {
+            for (const capability of capabilities) {
+                if (!capability['tb:options']) {
+                    capability['tb:options'] = {} as WebDriver.TestingbotCapabilities
+                }
 
-        tbOptions['tunnel-identifier'] = tbTunnelIdentifier
+                capability['tb:options']['tunnel-identifier'] = tbTunnelIdentifier
+            }
+        } else {
+            Object.keys(capabilities).forEach(browserName => {
```

Comment:
let's do a for..of loop instead
```suggestion
            for (const browserName of Object.keys(capabilities)) {
```